payments-core/src/main/java/com/stripe/android/model/PaymentMethodCreateParams.kt

```diff
@@ -174,52 +174,50 @@ data class PaymentMethodCreateParams internal constructor(
-        Fpx(PaymentMethod.Type.Fpx),
-        SepaDebit(PaymentMethod.Type.SepaDebit, hasMandate = true),
-        AuBecsDebit(PaymentMethod.Type.AuBecsDebit, hasMandate = true),
-        BacsDebit(PaymentMethod.Type.BacsDebit, hasMandate = true),
-        Sofort(PaymentMethod.Type.Sofort, hasMandate = true),
-        P24(PaymentMethod.Type.P24),
-        Bancontact(PaymentMethod.Type.Bancontact, hasMandate = true),
-        Giropay(PaymentMethod.Type.Giropay),
-        Eps(PaymentMethod.Type.Eps, hasMandate = true),
-        Oxxo(PaymentMethod.Type.Oxxo),
-        Alipay(PaymentMethod.Type.Alipay),
-        GrabPay(PaymentMethod.Type.GrabPay),
-        PayPal(PaymentMethod.Type.PayPal),
-        AfterpayClearpay(PaymentMethod.Type.AfterpayClearpay),
-        Upi(PaymentMethod.Type.Upi),
-        Netbanking(PaymentMethod.Type.Netbanking),
-        Blik(PaymentMethod.Type.Blik),
-        WeChatPay(PaymentMethod.Type.WeChatPay);
-
-        internal val code = paymentMethodType.code
-
-        companion object {
-            internal fun fromPaymentMethodType(paymentMethodType: PaymentMethod.Type) =
-                values().firstOrNull { it.paymentMethodType == paymentMethodType }
-        }
+    internal sealed class Type(
+        open val code: String,
+        open val hasMandate: Boolean = false
+    ) : Parcelable {
+        @Parcelize class Card : Type(PaymentMethod.Type.Card.code)
+        @Parcelize class Ideal : Type(PaymentMethod.Type.Ideal.code, hasMandate = true)
+        @Parcelize class Fpx : Type(PaymentMethod.Type.Fpx.code)
+        @Parcelize class SepaDebit : Type(PaymentMethod.Type.SepaDebit.code, hasMandate = true)
+        @Parcelize class AuBecsDebit : Type(PaymentMethod.Type.AuBecsDebit.code, hasMandate = true)
+        @Parcelize class BacsDebit : Type(PaymentMethod.Type.BacsDebit.code, hasMandate = true)
+        @Parcelize class Sofort : Type(PaymentMethod.Type.Sofort.code, hasMandate = true)
+        @Parcelize class P24 : Type(PaymentMethod.Type.P24.code)
+        @Parcelize class Bancontact : Type(PaymentMethod.Type.Bancontact.code, hasMandate = true)
+        @Parcelize class Giropay : Type(PaymentMethod.Type.Giropay.code)
+        @Parcelize class Eps : Type(PaymentMethod.Type.Eps.code, hasMandate = true)
+        @Parcelize class Oxxo : Type(PaymentMethod.Type.Oxxo.code)
+        @Parcelize class Alipay : Type(PaymentMethod.Type.Alipay.code)
+        @Parcelize class GrabPay : Type(PaymentMethod.Type.GrabPay.code)
+        @Parcelize class PayPal : Type(PaymentMethod.Type.PayPal.code)
+        @Parcelize class AfterpayClearpay : Type(PaymentMethod.Type.AfterpayClearpay.code)
+        @Parcelize class Upi : Type(PaymentMethod.Type.Upi.code)
+        @Parcelize class Netbanking : Type(PaymentMethod.Type.Netbanking.code)
+        @Parcelize class Blik : Type(PaymentMethod.Type.Blik.code)
+        @Parcelize class WeChatPay : Type(PaymentMethod.Type.WeChatPay.code)
```

Comment:
These can all be `object`

Reply:
They can't because of `@Parcelize`. Changed to make them static objects so they are equal if they are of the same type, as if they were enums.

Reply:
You can `@Parcelize` an `object`